be/src/agent/task_worker_pool.cpp

```diff
@@ -894,7 +902,12 @@ void* TaskWorkerPool::_publish_version_worker_thread_callback(void* arg_this) {
         // persist all related meta once in a group.
         tablets.reserve(tablet_ids.size());
         for (const auto tablet_id : tablet_ids) {
-            TabletSharedPtr tablet = StorageEngine::instance()->tablet_manager()->get_tablet(tablet_id);
+            auto st = StorageEngine::instance()->tablet_manager()->get_tablet(tablet_id);
+            if (!st.ok()) {
+                LOG(WARNING) << "failed to get tablet";
+                continue;
+            }
+            TabletSharedPtr tablet = st.value();
             if (tablet != nullptr) {
```

Comment:
According to your new implementation of `TabletManager::get_tablet()`, `st.value()` must not be NULL when `st.ok()` is true.